src/main/java/org/elasticsearch/index/search/child/TopChildrenQuery.java

```diff
@@ -176,11 +179,16 @@ int resolveParentDocuments(TopDocs topDocs, SearchContext context, ObjectObjectO
                 }
                 if (parentDocId != DocIdSetIterator.NO_MORE_DOCS) {
                     // we found a match, add it and break
-                    IntObjectOpenHashMap<ParentDoc> readerParentDocs = parentDocsPerReader.get(indexReader.getCoreCacheKey());
+                    IntObjectHashMap<ParentDoc> readerParentDocs = parentDocsPerReader.get(indexReader.getCoreCacheKey());
                     if (readerParentDocs == null) {
-                        //The number of docs in the reader and in the query both upper bound the size of parentDocsPerReader
-                        int mapSize =  Math.min(indexReader.maxDoc(), context.from() + context.size());
-                        readerParentDocs = new IntObjectOpenHashMap<>(mapSize);
+                        // TODO: this is the previous comment and code:
```

Comment:
I spotted this because the "expected elements" constructor now doesn't round up and throws an invalid argument exception. These values were [-1 + -1], as if the context was uninitialized at the time of construction. I think somebody ought to take a look at why it's so.